catalog/core/catalog-core-api/src/main/java/ddf/catalog/validation/AttributeValidatorRegistry.java

```diff
@@ -0,0 +1,52 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package ddf.catalog.validation;
+
+import java.util.Collection;
+import java.util.Set;
+
+/**
+ * Manages registered attribute validators.
+ * <p>
+ * <b> This code is experimental. While this interface is functional and tested, it may change or be
+ * removed in a future version of the library. </b>
+ * </p>
+ */
+public interface AttributeValidatorRegistry {
+    /**
+     * Registers the given {@link AttributeValidator}s to the attribute with the given name.
+     *
+     * @param attributeName the name of the attribute that the given validators will validate
+     * @param validators    the {@link AttributeValidator}s to apply to the attribute
+     * @throws IllegalArgumentException if {@code attributeName} is null, or if no
+     *                                  {@link AttributeValidator}s were passed
+     */
+    void registerValidators(String attributeName,
+            Collection<? extends AttributeValidator> validators);
+
+    /**
+     * Deregisters all {@link AttributeValidator}s associated with the given attribute.
+     *
+     * @param attributeName the name of the attribute whose validators will be deregistered
```

Comment:
For all those `null`, empty, missing value, return value, etc., comments, would you mind putting that information in the JavaDoc? Makes it easier for the clients and ensures that all the implementations are consistent.